t/api.t

```diff
@@ -117,5 +117,90 @@ subtest 'should combine' => sub {
 	is $secp->combine_public_keys($t{pubkey}, @to_combine), $combined_pubkey, 'combined pubkey ok';
 };
 
+subtest 'should sign and verify a message (recoverable)' => sub {
+	my $rec_sig = $secp->sign_message_recoverable($t{privkey}, $t{preimage});
+	ok defined($rec_sig), 'recoverable message signed ok';
+
+	# Test serialize compact
+	my $result = $secp->serialize_compact_recoverable($rec_sig);
+	ok defined($result->{signature}), 'compact signature extracted ok';
+	ok defined($result->{recovery_id}), 'recovery id extracted ok';
+	is length($result->{signature}), 64, 'compact signature length ok';
+	ok $result->{recovery_id} >= 0 && $result->{recovery_id} <= 3, 'recovery id range ok';
+
+	# Test recovery
+	my $recovered_pubkey = $secp->recover_public_key_message($rec_sig, $t{preimage});
+	is $recovered_pubkey, $t{pubkey}, 'public key recovered from message ok';
+
+	# Test verification methods
+	ok $secp->verify_message_recoverable($t{pubkey}, $rec_sig, $t{preimage}), 'recoverable message verified ok';
+
+	# Test with wrong pubkey should fail
+	my $wrong_pubkey = pack 'H*', '025476c2e83188368da1ff3e292e7acafcdb3566bb0ad253f62fc70f07aeee6358';
+	ok !$secp->verify_message_recoverable($wrong_pubkey, $rec_sig, $t{preimage}),
+		'wrong pubkey verification fails ok';
+};
+
+subtest 'should sign and verify a digest (recoverable)' => sub {
+	my $digest = sha256(sha256($t{preimage}));
+	my $rec_sig = $secp->sign_digest_recoverable($t{privkey}, $digest);
+	ok defined($rec_sig), 'recoverable digest signed ok';
+
+	# Test recovery
+	my $recovered_pubkey = $secp->recover_public_key($rec_sig, $digest);
+	is $recovered_pubkey, $t{pubkey}, 'public key recovered from digest ok';
+
+	# Test verification
+	ok $secp->verify_digest_recoverable($t{pubkey}, $rec_sig, $digest), 'recoverable digest verified ok';
+
+	# Test with bad digest should fail
+	my $bad_digest = $digest;
+	substr($bad_digest, 0, 1) = "\xff";
+	ok !$secp->verify_digest_recoverable($t{pubkey}, $rec_sig, $bad_digest),
+		'wrong digest verification fails ok';
+};
+
+subtest 'ethereum yellowpaper specification' => sub {
```

Comment:
I would appreciate a link to the specification here (as a comment)

Reply:
done